editor/components/EditableField.tsx

```diff
@@ -31,61 +25,72 @@ interface DataCardInputProps {
    * to set control the parent component's edit state
    */
   onCancel: () => void;
+
+  /** Validation rules that mirror react-hook-form RegisterOptions */
+  inputOptions?: RegisterOptions<FormValues, "value">;
+}
+
+interface FormValues {
+  value: string;
 }
 
 /**
  * Component that manages the form UI for a single
  * editable field
  */
-const DataCardInput = ({
+const EditableField = ({
   initialValue,
   inputType,
   isMutating,
   selectOptions,
   onSave,
   onCancel,
-}: DataCardInputProps) => {
-  // todo: input validation; input type = number
-  const [editValue, setEditValue] = useState<string>(initialValue);
-
-  const isDirty = editValue !== initialValue;
+  inputOptions,
+}: EditableFieldProps) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isDirty },
+  } = useForm<FormValues>({
+    defaultValues: {
+      value: initialValue,
+    },
+  });
+  
+  const onSubmit = async (data: FormValues) => {
+    await onSave(data.value);
+  };
 
   return (
-    <Form
-      onSubmit={async (e) => {
-        e.preventDefault();
-        await onSave(editValue);
-      }}
-    >
+    <Form onSubmit={handleSubmit(onSubmit)}>
       <div className="mb-2">
         {(inputType === "text" || inputType === "number") && (
           <Form.Control
+            {...register("value", inputOptions)}
```

Comment:
cool, i see how we're using the generic field name `"value"` here, and that matches the field name we established in the `FormValues` interface you created.

i got confused here briefly and wonder if a better name might be `"input_value"`?